packages/admin-panel/src/VizBuilderApp/components/PreviewSection.js

```diff
@@ -141,11 +163,12 @@ export const PreviewSection = () => {
       setFetchEnabled(false);
     },
     vizType,
+    previewMode: getTab(tab).previewMode,
```

Comment:
This has been a bug for a while, where we haven't been passing in the `previewMode` when previewing a viz, so we only ever got the data in array of objects structure.